src/base/http/responsebuilder.h

```diff
@@ -37,7 +37,7 @@ namespace Http
     {
     public:
         void status(uint code = 200, const QString &text = QLatin1String("OK"));
-        void header(const QString &name, const QString &value);
+        void setHeader(const Header &header);
```

Comment:
This method signature confuses. It creates the false impression that each subsequent call is applied to the same property (i.e. it resets a certain header).

Reply:
I don't feel the same way. There may be different `Header` objects, each `name` field could be different.

Reply:
Wouldn't `addHeader()` more appropriate name?

Reply:
I think it is not better. I would expect `addHeader()` to allow the same header to be added multiple times, but that is not possible with the current implementation.
https://github.com/qbittorrent/qBittorrent/pull/12793/files#diff-1a05bf5e5b8e2240e316d87f8c92b00a

Reply:
>I would expect addHeader() to allow the same header to be added multiple times

But it's meaningless in terms of subject domain to expect it...

Really the previous signature (with two parameters) clearly showed that you just set value for some particular header, isn't it?

However, as you wish... I don't have time to deal with WebAPI/UI related stuff so much...

Reply:
>But it's meaningless in terms of subject domain to expect it...

It is expected in some minor cases: https://stackoverflow.com/a/38837573

ps. PR is updated to address the other comment.